internal/services/recoveryservices/recovery_services_vault_hyperv_host_registration_key_datasource.go

```diff
@@ -0,0 +1,353 @@
+	return vaultcertificates.ValidateCertificateID
+}
+
+func (h HyperVHostRegistrationKeyDataSource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"site_recovery_services_vault_hyperv_site_id": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: replicationfabrics.ValidateReplicationFabricID,
+		},
+
+		"validate_in_hours": {
+			Type:         schema.TypeInt,
+			ForceNew:     true,
+			Optional:     true,
+			Default:      120,
+			ValidateFunc: validation.IntAtLeast(0),
+		},
+	}
+}
+
+func (h HyperVHostRegistrationKeyDataSource) Attributes() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"xml_content": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"resource_id": {
+			Type:     schema.TypeInt,
+			Computed: true,
+		},
+		"management_cert": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"aad_tenant_id": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"aad_authority": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"service_principal_client_id": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"aad_vault_audience": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"aad_management_endpoint": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"vault_private_endpoint_state": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+		"validate_to": {
+			Type:     schema.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (h HyperVHostRegistrationKeyDataSource) ModelObject() interface{} {
+	return &HyperVHostRegistrationKeyModel{}
+}
+
+func (h HyperVHostRegistrationKeyDataSource) ResourceType() string {
+	return "azurerm_recovery_services_vault_hyperv_host_registration_key"
+}
+
+func (h HyperVHostRegistrationKeyDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var metaModel HyperVHostRegistrationKeyModel
+			if err := metadata.Decode(&metaModel); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			FabricId, err := replicationfabrics.ParseReplicationFabricID(metaModel.HyperVSiteId)
+			if err != nil {
+				return fmt.Errorf("parsing: %+v", err)
+			}
+
+			date := time.Now().Format("2-01-2006")
+			name := fmt.Sprintf(`CN=CB_%s-%s-vaultcredentials`, FabricId.VaultName, date)
+
+			id := vaultcertificates.NewCertificateID(FabricId.SubscriptionId, FabricId.ResourceGroupName, FabricId.VaultName, name)
+
+			metadata.SetID(id)
+
+			return resourceRecoveryServicesVaultHyperVHostRegistrationKeyCreateInternal(ctx, id, metadata)
+		},
+	}
+}
+
+func resourceRecoveryServicesVaultHyperVHostRegistrationKeyCreateInternal(ctx context.Context, id vaultcertificates.CertificateId, metadata sdk.ResourceMetaData) error {
```

Comment:
Since this is a data source we shouldn't be creating anything, what is the purpose of this data source?

Reply:
It's used to register HyperV host to Recovery Vault, could get the same file on Portal: Any Recovery Services Vault -> Site Recovery infrastructure -> HyperV Hosts -> Add Server: on the step 4, the "vault registration key" is created by calling the same API and save to a file. 
<img width="248" alt="WeChat38b74ebf493450b7b499b249370053da" src="https://user-images.githubusercontent.com/51212351/220576251-05770a1a-c0aa-447d-bc44-b32f8287734d.png">

I have thought if I should use Resource rather than Data Source, while... it does not expose anything to read...